Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -47,25 +48,28 @@ bool LogFileReader::readAll(const QString& fileName)
             emit batteryDataReceived(batteryData);
         }
     }
-
     return true;
 }
 
-/* File input is a csv file in the format of hh:mm:ss:zzz, voltage, current.
- * Negative current values denote a charging battery.
- * Need to implement error checking for the correct number of values and
- * that the conversion from string to double is sucessful.*/
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
     QStringList sections = line.split(BATDATA_DELIMITER);
-
+    if(sections.size() != 3 || sections.at(0).isNull() || sections.at(1).isNull() || sections.at(2).isNull())
+    {
+        return false;
+    }
     QString timeString = sections.at(0);
     batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
+    if(!batteryData.time.isValid())
+        return false;
 
-    batteryData.voltage = sections.at(1).toDouble();
-
-    batteryData.current = sections.at(2).toDouble();
+    bool voltageDoubleFail;
```

Comment:
This naming is a little weird; right now we have it so if if voltageDoubleFail is true..... then the voltage double is fine, I would rename this to be along the lines of `voltageOk`. Same with `currentOk`